feature-libs/product-configurator/rulebased/components/default-form/configurator-default-form.component.ts

```diff
@@ -0,0 +1,139 @@
+  Component,
+  Input,
+  OnInit,
+} from '@angular/core';
+import { LanguageService } from '@spartacus/core';
+import {
+  CommonConfigurator,
+  ConfiguratorRouterExtractorService,
+} from '@spartacus/product-configurator/common';
+import { Observable } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { ConfiguratorCommonsService } from '../../core/facade/configurator-commons.service';
+import { ConfiguratorGroupsService } from '../../core/facade/configurator-groups.service';
+import { Configurator } from '../../core/model/configurator.model';
+import { ConfiguratorStorefrontUtilsService } from '../service/configurator-storefront-utils.service';
+import { ConfigFormUpdateEvent } from './configurator-default-form.event';
+import { ConfiguratorExpertModeService } from '../../core/services/configurator-expert-mode.service';
+
+@Component({
+  selector: 'cx-configurator-default-form',
+  templateUrl: './configurator-default-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfiguratorDefaultFormComponent implements OnInit {
+  @Input() group: Configurator.Group;
+  @Input() owner: CommonConfigurator.Owner;
+  @Input() isNavigationToGroupEnabled: boolean;
+
+  activeLanguage$: Observable<string> = this.languageService.getActive();
+  uiType = Configurator.UiType;
+
+  constructor(
+    protected configuratorCommonsService: ConfiguratorCommonsService,
+    protected configuratorGroupsService: ConfiguratorGroupsService,
+    protected configRouterExtractorService: ConfiguratorRouterExtractorService,
+    protected languageService: LanguageService,
+    protected configUtils: ConfiguratorStorefrontUtilsService,
+    protected configExpertModeService: ConfiguratorExpertModeService
+  ) {}
+
+  ngOnInit(): void {
+    this.configRouterExtractorService
+      .extractRouterData()
+      .pipe(take(1))
+      .subscribe((routingData) => {
+        //In case of resolving issues, check if the configuration contains conflicts,
+        //if not, check if the configuration contains missing mandatory fields and show the group
+        if (routingData.resolveIssues) {
+          this.configuratorCommonsService
+            .hasConflicts(routingData.owner)
+            .pipe(take(1))
+            .subscribe((hasConflicts) => {
+              if (hasConflicts && !routingData.skipConflicts) {
+                this.configuratorGroupsService.navigateToConflictSolver(
+                  routingData.owner
+                );
+
+                //Only check for Incomplete group when there are no conflicts or conflicts should be skipped
+              } else {
+                this.configuratorGroupsService.navigateToFirstIncompleteGroup(
+                  routingData.owner
+                );
+              }
+            });
+        }
+        if (routingData.expMode) {
+          this.configExpertModeService?.setExpModeRequested(
+            routingData.expMode
+          );
+        }
+      });
+  }
```

Comment:
@Larisa-Staroverova: i believe this code makes only sense in context of the form-component, but not the dialog-solver-component, so i suggest to move it to the form-component.

Reply:
Agree!

Reply:
Done!